scene/gui/split_container.cpp

```diff
@@ -95,28 +96,61 @@ void SplitContainerDragger::_notification(int p_what) {
 	switch (p_what) {
 		case NOTIFICATION_MOUSE_ENTER: {
 			mouse_inside = true;
-			SplitContainer *sc = Object::cast_to<SplitContainer>(get_parent());
-			if (sc->get_theme_constant(SNAME("autohide"))) {
-				queue_redraw();
-			}
+			queue_redraw();
 		} break;
 
 		case NOTIFICATION_MOUSE_EXIT: {
 			mouse_inside = false;
-			SplitContainer *sc = Object::cast_to<SplitContainer>(get_parent());
-			if (sc->get_theme_constant(SNAME("autohide"))) {
-				queue_redraw();
-			}
+			queue_redraw();
 		} break;
 
 		case NOTIFICATION_DRAW: {
 			SplitContainer *sc = Object::cast_to<SplitContainer>(get_parent());
-			if (!dragging && !mouse_inside && sc->get_theme_constant(SNAME("autohide"))) {
+
+			Ref<StyleBox> bg_sb = NULL;
+			if (dragging) {
+				bg_sb = sc->_get_bar_stylebox_pressed();
+			} else if (mouse_inside) {
+				bg_sb = sc->_get_bar_stylebox_hover();
+			} else {
+				bg_sb = sc->_get_bar_stylebox();
+			}
+
+			int inner_bar_thickness = sc->theme_cache.bar_thickness;
+			if (inner_bar_thickness == -1) {
+				inner_bar_thickness = sc->theme_cache.separation;
+			}
+			Rect2 inner_bar_rect = Rect2();
+			if (sc->vertical) {
+				inner_bar_rect = Rect2(Point2(0, int((get_size().height - inner_bar_thickness) / 2)), Size2(get_size().width, inner_bar_thickness));
+			} else {
+				inner_bar_rect = Rect2(Point2(int((get_size().width - inner_bar_thickness) / 2), 0), Size2(inner_bar_thickness, get_size().height));
+			}
+
+			draw_style_box(bg_sb, inner_bar_rect);
+
+			if (dragging) {
+				if (sc->get_theme_constant(SNAME("grabber_hidden_pressed"))) {
```

Comment:
This should be replaced with use of `theme_cache`. You can make SplitContainerDragger a friend class of SplitContainer.